src/cljs/main/broadfcui/main.cljs

```diff
@@ -120,68 +120,106 @@
   {:get-initial-state
    (fn []
-     {:failedRetries 0})
+     {:failed-retries 0})
    :render
    (fn [{:keys [this state]}]
-     (when-let [alerts (not-empty (:alerts @state))]
-       [:div {}
+     (let [service-alerts (:service-alerts @state) js-alerts (:js-alerts @state)]
+       ; Why was zIndex of 514 chosen? IGV has a zIndex of 512, so modals use a zIndex of 513
+       ; We want these banners to be shown in front of the modals, so we use a zIndex of 514
+       [:div {:style {:zIndex 514 :position "relative"}}
+        (map (fn [alert]
+               [Banner {:title (:title alert)
+                        :background-color (:exception-state style/colors)
+                        :text-color "#fff"
+                        :message (:message alert)
+                        :link [:a {:href (str (:link alert)) :target "_blank" :style {:color "#fff"}} "Read more..."]}])
+             service-alerts)
         (map (fn [alert]
-               (status-alert (:title alert) (:message alert) (:link alert)))
-             alerts)]))
+               [Banner {:title "Uh oh! Something may be wrong..."
+                        :background-color (:warning-state style/colors)
+                        :text-color "#000"
+                        :message "There was an error in FireCloud. It may not mean anything, but you should consider reloading the page to be safe."
+                        :more-content [:div {:style {:paddingTop "0.5rem"}}
+                                       [:div {} "Here are some details about the error that occurred. If you post this on our "
+                                        [:a {:href "http://gatkforums.broadinstitute.org/firecloud/categories/ask-the-firecloud-team"
+                                             :target "_blank" :style {}} "forum"] ", our team can take a look."]
+                                       [:div {:style {:paddingTop "0.5rem"}} [:div {:style {:fontWeight "bold"}} "Stack trace: "]]
+                                       (map (fn [line]
+                                              [:div {} line])
+                                            (clojure.string/split-lines (:stack alert)))
+                                       [:div {:style {:paddingTop "0.5rem"}} [:div {:style {:fontWeight "bold"}} "Source: "] (:source alert)]]}])
+             js-alerts)]))
    :component-did-update
    (fn [{:keys [this state locals]}]
      ;; Reset the interval
      (js/clearInterval (:interval-id @locals))
      ;; Update the poll interval based on the number of failed attempts (for exponential back offs)
      (swap! locals assoc :interval-id
-            (js/setInterval #(this :load-alerts) (status-alert-interval (:failedRetries @state)))))
+            (js/setInterval #(this :load-service-alerts) (status-alert-interval (:failed-retries @state)))))
    :component-did-mount
    (fn [{:keys [this state locals]}]
+     ;; Set the js error listener
+     (set! js/window.onerror
```

Comment:
Also, since this is a globally shared thing, any place in the code could set this and accidentally overwrite this one. A convention I've been using is to have the `App` touch stuff like this. That component could set this and then delegate the handling to someone else.